src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -415,6 +417,7 @@ private static string UrlDecodeInternal(string value, Encoding encoding)
 
                         // don't add as char
                         helper.AddByte(b);
+                        needsDecoding = true;
```

Comment:
According to the code coverage report, there are missed branches for this if.  Can you please add tests to ensure it's fully covered?